src/drawers/circleDrawer.ts

```diff
@@ -12,6 +12,27 @@ export module _Drawer {
     public _getPixelPoint(datum: any, index: number): Point {
       return { x: this._attrToProjector["cx"](datum, index), y: this._attrToProjector["cy"](datum, index) };
     }
+
+    public _getSelectionDistance(selection: D3.Selection, pixelPoint: Point): number {
+      var circleX = parseFloat(selection.attr("cx"));
+      var circleY = parseFloat(selection.attr("cy"));
+      var circleRadius = parseFloat(selection.attr("r"));
+
+      var circleCenter = {x: circleX, y: circleY};
+      if (_Util.Methods.pointDistance(circleCenter, pixelPoint) <= circleRadius) {
+        return 0;
+      } else {
+        var angle = Math.atan2(pixelPoint.y - circleY, pixelPoint.x - circleX);
+        var closestPoint =  { x: circleX + circleRadius * Math.cos(angle), y: circleY + circleRadius * Math.sin(angle) };
+        return _Util.Methods.pointDistance(pixelPoint, closestPoint);
```

Comment:
I see where you're getting at.  Let me try to clean the code with a similar approach.